src/net-utils.ts

```diff
@@ -48,14 +58,55 @@ async function isPortOpen(
  */
 export async function openBrowserWhenReady(
   port: number,
-  ...additionalPorts: number[]
+  additionalPorts: number[] = [],
+  terminal?: vscode.Terminal
 ): Promise<void> {
-  const portsOpen = [port, ...additionalPorts].map((p) =>
-    retryUntilTimeout(10000, () => isPortOpen("127.0.0.1", p))
+  const portsOpenResult = await vscode.window.withProgress(
+    {
+      location: vscode.ProgressLocation.Notification,
+      title: "Waiting for Shiny app to start...",
+      cancellable: false,
+    },
+    async () => {
+      const portsOpen = [port, ...additionalPorts].map((p) =>
+        retryUntilTimeout(10000, () => isPortOpen("127.0.0.1", p))
```

Comment:
Do you mind if we make this at least 20 seconds? Someone was already having trouble on Discord from this not being long enough. I get waits of that long if I've freshly installed big packages and they need to byte compile, for example.

Reply:
No problem!

```suggestion
        retryUntilTimeout(20000, () => isPortOpen("127.0.0.1", p))
```